### ISSUE TITLE
Loss function 

### ISSUE THREAD

**Issue by tavildar** [hienpham15/CNN_model_ColorConstancy#5]
Hi, 

Got here from the FC4 page. As per my understanding, the loss function used in FC4 is 

> acos( sum(l2_normalized(y_true) * l2_normalized(y_pred))). 

You are using cosine proximity which is missing the cos inverse (acos). See [here](https://github.com/keras-team/keras/blob/master/keras/losses.py)

**Comment by tavildar** [comment]
Hi, cosine_proximity is not in radians. You are missing the arccos which converts it to radians. I agree either of them is a reasonable loss function, but they are not equivalent. Anyway, just a suggestion from my side. Feel free to close the issue.

**Comment by hienpham15** [comment]
@tavildar 
Thanks for pointing out my misunderstanding, I tried your suggestion and it did improve the performance of the model.

**Comment by tavildar** [comment]
Great. Good to know, and thanks for your code!
